internal/services/containerapps/container_app_resource.go

```diff
@@ -409,7 +409,51 @@ func (r ContainerAppResource) Update() sdk.ResourceFunc {
 func (r ContainerAppResource) CustomizeDiff() sdk.ResourceFunc {
 	return sdk.ResourceFunc{
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
-			if metadata.ResourceDiff != nil && metadata.ResourceDiff.HasChange("secret") {
+			if metadata.ResourceDiff == nil {
+				return nil
+			}
+			var app ContainerAppModel
+			if err := metadata.DecodeDiff(&app); err != nil {
+				return err
+			}
+			// Ingress traffic weight validations
+			if len(app.Ingress) != 0 {
+				ingress := app.Ingress[0]
+				if old, _ := metadata.ResourceDiff.GetChange("name"); old == "" {
```

Comment:
During the recreation (after tainted), the `name` is also marked as to be changed, from "some value" to null?

Reply:
By experimenting, it turns out the first invocation on CustomizeDiff for force new resource will compare the state and the config (instead of state vs null). So you are right, and I've updated the code accordingly.

Whilst one limitation to this resource is that, when users update the container app to have multiple ingress traffic weight, then they can't simply rename the resource to re-provision. The validation will fail and they'll have to manually ensure there is only one traffic weight defined as expected.